### PR TITLE
chore: create semgrep.yml

### PR DESCRIPTION
**Pull request by hrushikeshdeshpande**
Description - Adding Semgrep.yml file as part of App&ProdSec team's initiative to onboard Semgrep for Secrets scanning and more.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: this is a CI change
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: this is a CI change
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Changeset included
  - [x] Changeset not necessary because: this is a CI job in itself
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: internal change

### CODE REVIEW

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+on:
+  pull_request: {}
+  workflow_dispatch: {}
+  push: 
+    branches:
+      - main
+      - master
```
Comment (penalosa) [inline .github/workflows/semgrep.yml:7]:
```suggestion
```

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (penalosa) [inline .github/workflows/semgrep.yml:9]: Is this needed if it also runs on `main`?

Comment (hrushikeshdeshpande) [reply .github/workflows/semgrep.yml:9]: It will run on main when a PR is merged and also daily. This is how we have configured it on bitbucket as well.

**Review by emily-shen — APPROVED** [review]

**Comment by vicb** [comment]
It looks like the CI is failing since this change was introduced

**Pull request merged**